### PR TITLE
Email connector action

### LINKED ISSUE #6: Connector step emit email
Title, to, cc, body are all mustache templates and / or atto editor with placeholders expressions

Some vars:
* rows process from source
* rows processed at this loader
* time started
* time finished
* other variables in the "global" scope of the stream rather than each item

### LINKED ISSUE #410: Make a new var which holds a deep link to the run instance summary page
Split from https://github.com/catalyst/moodle-tool_dataflows/issues/408

### PR DESCRIPTION
**Pull request by brendanheywood**
Resolves #410 
Resolves #6

### CODE REVIEW

File: classes/local/step/connector_email.php [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        }
+
+        $config = $this->enginestep->stepdef->config;
+
+        $toemail = $config->to;
+        $to = (object)array(
+            'id' => 1,
+            'auth' => 'manual',
+            'email' => $toemail,
+            'username' => 'xxx',
```
Comment (brendanheywood) [inline classes/local/step/connector_email.php:78]: Need to fix this

Comment (keevan) [reply classes/local/step/connector_email.php:78]: On that note, does this support multiple addresses? e.g. if it wasn't for a single user but for a fixed number of predefined users instead (which wasn't sourced in Moodle itself)?

Comment (brendanheywood) [reply classes/local/step/connector_email.php:78]: No and at this point I'm happy for that to be out of scope and tackled later on

**Comment by brendanheywood** [comment]
Example email flow:

[Send_some_emails!_20220805_1443.yml.txt](https://github.com/catalyst/moodle-tool_dataflows/files/9269626/Send_some_emails._20220805_1443.yml.txt)

**Pull request merged**